### ISSUE TITLE
Digest is not optional anymore

### ISSUE THREAD

**Issue by b4nst** [stefanprodan/timoni#175]
The new `minimal` module introduced sort of a breaking change. Before it, image digest [used to be optional](https://github.com/stefanprodan/timoni/blob/c9f0c1e4f34c9065b2f9418b5a9f2442b50c8e42/examples/podinfo/templates/config.cue#L39C19-L39C19). In the new [#Image definition](https://github.com/stefanprodan/timoni/blob/c9f0c1e4f34c9065b2f9418b5a9f2442b50c8e42/examples/minimal/cue.mod/pkg/timoni.sh/core/v1alpha1/image.cue#L8) (thanks for that btw) it is not optional anymore. Is that on purpose?

If I remove it from the `values.cue`:

```
timoni mod lint .
4:24PM ERR build failed: timoni.apply.all.0.metadata.labels: 2 errors in empty disjunction: (and 2 more errors)
make: *** [lint] Error 1
```

Also is there a way to have a better error message? Took me a while to find the root cause 😅. I know by experience it's not that obvious with the CUE API, I can also try some tricks on my side and push you a PR if I find any solution.

**Comment by stefanprodan** [comment]
Yes digest is now required, but it can be set to string empty e.g. `digest: ""`. The goal is to force people into using immutable images, they can still use just tags, but they must opt-out by setting digest to string empty.

**Comment by stefanprodan** [comment]
> Also is there a way to have a better error message? 

I hope so, I've upgrade Timoni to CUE 0.6.0 in main, so we can now use required fields e.g. `digest!`. Marking all the config required fields with `!` should yell better error messages.

**Comment by b4nst** [comment]
> The goal is to force people into using immutable images, they can still use just tags, but they must opt-out by setting digest to string empty.

Copy that, fully agree with the reason so I'll go ahead and close this ticket. Thanks!

**Comment by stefanprodan** [comment]
@b4nst with #176 the error messages are better:

```
6:20PM ERR build failed: timoni.apply.all.3._config.image: cannot reference optional field: digest
```

Guess `cannot reference optional field` should be `cannot reference required field`, maybe a CUE bug.
